### PR TITLE
BE17 Split Controller

### LINKED ISSUE #17: Split Controller into Overlay, Extension and TwitchBot

### PR DESCRIPTION
**Pull request by louisheal**
split controller into twitch bot, overlay and extension
move test files next to the files they test
move interfaces to the consumer

### CODE REVIEW

File: services/auth.go [hunk 1]
```diff
@@ -24,28 +23,26 @@ type Receipt struct {
...
-	VerifyReceipt(tokenString string) (*Receipt, error)
+type AuthService struct {
+	channelRepo  ChannelRepo
+	clientSecret string
 }
 
-type authService struct {
-	channelRepo  repositories.ChannelRepo
-	clientSecret string
+type ChannelRepo interface {
```
Comment (louisheal) [inline services/auth.go:31]: Rename

File: controllers/extension.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	return &ExtensionController{
+		Announcer: announcer,
+		Verifier:  verifier,
+		Store:     store,
+		Users:     users,
+	}
+}
+
+func (c *ExtensionController) GetStoreData(ctx *gin.Context) {
+	tokenString := ctx.GetHeader("x-extension-jwt")
```
Comment (louisheal) [inline controllers/extension.go]: Should "x-extension-jwt" be a const?

File: controllers/twitch_bot.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	ItemGetterSetter
+}
+
+type UserIDGetter interface {
+	GetUserID(username string) (models.TwitchID, error)
+}
+
+type TwitchBotController struct {
+	Announcer Announcer
+	Database  ViewerItemSetterGetter
```
Comment (louisheal) [inline controllers/twitch_bot.go:37]: Should this become two repositories?
e.g. ItemRepo + ViewerRepo

My reasoning for not doing it before is that a "viewer" doesn't exist in the database.
Additionally, to retrieve a Viewer, we need to retrieve their selected item, so the viewer repository would need to contain the item repository.

(A Viewer is an instance of a User within a specific Channel)

File: controllers/extension.go (hunk 2 above)
Comment (RAshkettle) [reply controllers/extension.go]: If you were using it in more than one place, I'd say always.  For a single instance, it's not worth the effort.

File: controllers/twitch_bot.go (hunk 3 above)
Comment (RAshkettle) [reply controllers/twitch_bot.go:37]: Splitting them now will save you a big headache in the future.  You can always put common code in its own package that both use.

Comment (louisheal) [reply controllers/twitch_bot.go:37]: This will be done in https://github.com/StreamPets/backend/issues/20

**Pull request merged**